src/pseudopeople/loader.py

```diff
@@ -0,0 +1,114 @@
+
+    # Load data
+    df_1040 = load_standard_dataset_file(data_path[DatasetNames.TAXES_1040], user_filters)
+    df_dependents = load_standard_dataset_file(
+        data_path[DatasetNames.TAXES_DEPENDENTS], user_filters
+    )
+
+    # Get wide format of dependents - metadata for each guardian's dependents
+    dependents_wide = flatten_data(
+        data=df_dependents,
+        index_col=COLUMNS.guardian_id.name,
+        rank_col=COLUMNS.simulant_id.name,
+        value_cols=[
+            COLUMNS.simulant_id.name,
+            COLUMNS.first_name.name,
+            COLUMNS.last_name.name,
+            COLUMNS.ssn.name,
+        ],
+    )
+    # Rename tax_dependents columns
+    dependents_wide = dependents_wide.add_prefix("dependent_").reset_index()
+    # Widen 1040 data (make one row for spouses that are joint filing)
+    df_joint_1040 = combine_joint_filers(df_1040)
+
+    # Merge tax dependents onto their guardians - we must do it twice, merge onto each spouse if joint filing
+    tax_1040_w_dependents = df_joint_1040.merge(
+        dependents_wide,
+        how="left",
+        left_on=COLUMNS.simulant_id.name,
+        right_on=COLUMNS.guardian_id.name,
+    )
+    # todo: uncomment with mic-4244. Handle columns with dependents for both guardians
+    # tax_1040_w_dependents = tax_1040_w_dependents.merge(dependents_wide, how="left", left_on="COLUMNS.spouse_simulant_id.name", right_on="COLUMNS.guardian_id.name")
+
+    return tax_1040_w_dependents
+
+
+def flatten_data(
+    data: pd.DataFrame,
+    index_col: str,
+    rank_col: str,
+    value_cols: List[str],
+    ascending: bool = False,
+) -> pd.DataFrame:
+    # Function that takes a dataset and widens (pivots) it to capture multiple metadata columns
+    # Example: simulant_id, dependdent_1, dependent_2, dependent_1_name, dependent_2_name, etc...
+    data = data.copy()
+    data["rank"] = (
+        data.groupby(index_col, group_keys=False)[rank_col]
+        .apply(lambda x: x.rank(method="first", ascending=ascending))
+        .astype(int)
+    )
+    # todo: Improve via mic-4244 for random sampling of dependents
+    # Choose 4 dependents
+    data = data.loc[data["rank"] < 5]
+    data["rank"] = data["rank"].astype(str)
+    flat = data.pivot(columns="rank", index=index_col, values=value_cols)
+    flat.columns = ["_".join([pair[1], pair[0]]) for pair in flat.columns]
+    return flat
+
+
+def combine_joint_filers(data: pd.DataFrame) -> pd.DataFrame:
+    # Get groups
+    joint_filers = data.loc[data["joint_filer"] == True]
+    reference_persons = data.loc[
+        data[COLUMNS.relation_to_reference_person.name] == "Reference person"
+    ]
+    no_spouses_index = data.index.difference(
+        joint_filers.index.union(reference_persons.index)
+    )
+    no_spouses = data.loc[no_spouses_index]
+
+    joint_filers = joint_filers.add_prefix("spouse_")
+    # Merge spouses
+    spouses = reference_persons.merge(
```

Comment:
This is also a bit confusing, since this should contain all filers who are the reference person, whether or not they have a spouse (and whether or not their spouse filed jointly)

Reply:
Yes. This is all reference persons with and without a joint filer. Another bad variable name.